### PR TITLE
Add SQS event integration

### PR DESCRIPTION
**Pull request by alexdebrie**
## What did you implement:

Adds SQS event integration 🎉 

## How did you implement it:

EDIT 6/28: SQS integration is live! Full testing instructions below.

I saw [Brett Andrews note that Lambda + SQS integration is coming soon](https://twitter.com/AWSbrett/status/1010008780375928833). Then I peeked at the [SAM implementation](https://github.com/awslabs/serverless-application-model/pull/451) of the SQS event. It looks to be very similar to Kinesis / DynamoDB streams as it's a pull-based integration. Like the streams, you can set batch size. There is no need for the `startingPosition` parameter though.

I found the necessary IAM permissions by looking at the permissions on [this role](https://github.com/awslabs/serverless-application-model/pull/477/files#diff-dc24340def2e9285840c922890edac32R8).

## How can we verify it:

We need to wait until the integration is actually released. Once it is, you could use the following `serverless.yml` to test:

```yml
# serverless.yml

service: sqs

provider:
  name: aws
  runtime: nodejs6.10

functions:
  hello:
    handler: handler.hello
    events:
      - sqs:
          arn:
            Fn::GetAtt:
              - MyQueue
              - Arn

resources:
  Resources:
    MyQueue:
      Type: "AWS::SQS::Queue"
      Properties:
        QueueName: "MyQueue"
```

After you deploy, run the following commands with the `awscli` to put a message into your Queue:

```bash
$ QUEUE_URL=`aws sqs get-queue-url --queue-name MyQueue --query 'QueueUrl' --output text`
$ aws sqs send-message --queue-url ${QUEUE_URL} --message-body "test"
```

Then check to make sure your Lambda fired:

```bash
$ sls metrics
Service wide metrics
June 27, 2018 3:16 PM - June 28, 2018 3:16 PM

Invocations: 1
Throttles: 0
Errors: 0
Duration (avg.): 3.63ms
```

## Todos:

- [X] Write tests
- [X] Write documentation
- [X] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [X] Provide verification config / commands / resources
- [X] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: lib/plugins/Plugins.json [hunk 1]
```diff
@@ -46,6 +46,7 @@
     "./aws/package/compile/events/cloudWatchEvent/index.js",
     "./aws/package/compile/events/cloudWatchLog/index.js",
     "./aws/package/compile/events/cognitoUserPool/index.js",
+    "./aws/package/compile/events/sqs/index.js",
```
Comment (jeremydaly) [inline lib/plugins/Plugins.json]: I was wondering why this plugin wasn't running!

File: docs/providers/aws/events/sqs.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+For the SQS event integration, you can set the `batchSize`, which effects how many SQS messages will be included in a single Lambda invocation. The default `batchSize` is 10.
+
+```yml
+functions:
+  preprocess:
+    handler: handler.preprocess
+    events:
+      - sqs:
+          arn: arn:aws:sqs:region:XXXXXX:myQueue
+          batchSize: 100
```
Comment (jeremydaly) [inline docs/providers/aws/events/sqs.md]: 10 is the maximum batch size allowed

File: lib/plugins/Plugins.json (hunk 1 above)
Comment (alexdebrie) [reply lib/plugins/Plugins.json]: 🙈 Yea, took me a while to find this one 😁

File: docs/providers/aws/events/sqs.md (hunk 2 above)
Comment (alexdebrie) [reply docs/providers/aws/events/sqs.md]: Thanks! Updated.

**Comment by domaslasauskas** [comment]
It's here! 🎉 [AWS Lambda Adds Amazon Simple Queue Service to Supported Event Sources](
https://aws.amazon.com/blogs/aws/aws-lambda-adds-amazon-simple-queue-service-to-supported-event-sources/)

**Comment by pmuens** [comment]
This is awesome! Thanks for implementing this @alexdebrie 👍 🎉

**Comment by QAnders** [comment]
Any plans to implement the "Extended Client" functionality as in AWS Java SDK to support messages larger than 256kb by storing the payload/body on S3?

E.g.:

```
resources:
  Resources:
    MyQueue:
      Type: "AWS::SQS::Queue"
      Properties:
        ExtendedLargeMsg: "MyS3Bucket"
        QueueName: "MyQueue"
```

**Pull request merged**